### PR TITLE
Team channels widget storybook

### PR DESCRIPTION
**Pull request by buoyad**
- Make `teams/common` into a dir
- Add `extraData` param to `usePopup` so you can trigger updating the popup even if `showingPopup` doesn't change.
- Add `useAutocompleter` to use on desktop - shows the top five options given a filter and handles keyboard controls and clicks.

image

cc @keybase/y2ksquad

### CODE REVIEW

**Comment by buoyad** [comment]
- [ ] storyshots before merge

File: shared/teams/common/channels-widget.tsx [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    onAdd(value)
+    setFilter('')
+  }
+
+  const {popup, popupAnchor, onKeyDown, setShowingPopup} = useAutocompleter(channelnames, onSelect, filter)
+
+  return (
+    <>
+      <Kb.SearchFilter
+        // @ts-ignore ref typing
```
Comment (jakob223) [inline shared/teams/common/channels-widget.tsx]: this shouldn't have to be `ts-ignore`d. Possibly `Kb.SearchFilter` needs to be forwarding a ref to a subcomponent, or possibly you need to use `usePopup<Kb.SearchFilter>`

File: shared/teams/common/use-autocompleter.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            >
+              <Kb.Text type="BodySemibold" lineClamp={1}>
+                {item}
+              </Kb.Text>
+            </Kb.Box2>
+          </Kb.ClickableBox>
+        ))}
+      </Kb.Overlay>
+    ),
+    filterLCase + selected
```
Comment (jakob223) [inline shared/teams/common/use-autocompleter.tsx]: might be cool if this were an array and worked like `deps` does for other hooks.

File: shared/teams/common/channels-widget.tsx [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+  const [filter, setFilter] = React.useState('')
+  const channels = Container.useSelector(s => Constants.getTeamChannelInfos(s, teamID))
+  const channelnames = [...channels.values()].map(c => `#${c.channelname}`)
+
+  const onSelect = value => {
+    onAdd(value)
+    setFilter('')
+  }
+
+  const {popup, popupAnchor, onKeyDown, setShowingPopup} = useAutocompleter(channelnames, onSelect, filter)
```
Comment (jakob223) [inline shared/teams/common/channels-widget.tsx]: perhaps we should only include channels in the autocompleter that aren't already chosen? it seems weird that `#general` is the first suggestion.

**Pull request merged**